utility/map_str_enhancer.go

```diff
@@ -15,7 +16,7 @@ func Add(m common.MapStr, key string, val interface{}) {
 		return
 	}
 
-	switch val.(type) {
+	switch value := val.(type) {
 	case *bool:
 		if newVal := val.(*bool); newVal != nil {
```

Comment:
holding off on refactoring in this pr to use `value` and avoid double type assertion, will do that next